pkg/operator/ceph/controller/spec.go

```diff
@@ -645,6 +681,22 @@ func (c *daemonConfig) buildAdminSocketCommand() string {
 	return command
 }
 
+func (c *daemonConfig) buildDaemonReadinessCheckCommand() (string, string) {
+	command := ""
+	msg := ""
+	if c.daemonType == config.MgrType {
+		// As per the current ceph mgr implementation, only active manager
+		// offers the "mgr_status" command. Standby interface doesn't support
+		// this command.
+		command = "mgr_status"
+		msg = "Standby manager does not process requests."
+	} else {
+		logger.Errorf("Cannot generate readiness probe: Unsupported daemon %s.", config.MgrType)
```

Comment:
If all we do is log an error, we may miss the problem. If the mgr is the only one we're implementing it in for now, we should either just implement it for the mgr (and not generalize the helper), or we would need to return an error and fail the reconcile. I'm thinking it should just be implemented for the mgr specifically, and avoid the error condition since this is something we can fully catch during development, and no need for a runtime error.

Reply:
@travisn I factored out the ProbeHandler and made the readiness generation more specefic (tailored to the mgr). In case more daemons in the future need something similar then we can make the code more generic.